tst/JsonApiTest.php

```diff
@@ -325,6 +325,9 @@ public function baseUriProvider()
      */
     public function testShortenViaYourlsFailure()
```

Comment:
Okay this test tests a different error, does not it?

If so, can we maybe also have a real API (integration IMHO) test that tests this?

Because one thing I am not yet sure about is URL encoding: Does the issue only happen when the URL is not properly encoded (as in `tst/YourlsProxyTest.php`)? I would guess so?
Does this mean anything for us?

Reply:
The contents of the $_GET array are already decoded (by PHP) - in the tests we can therefore set them to the decoded values.

This test happened to work because of the bug: The empty string of $options['main']['basepath'] was always found, meaning `[...] === false` was never true. Now we have to be precise and configure the basepath to still have it detect that expected failure.

Reply:
The `REQUEST_URI` is needed to be set for the `yourlsproxy` operation to get set:
https://github.com/PrivateBin/PrivateBin/blob/8e6e31db5cd7e50cfc6c1eeece8ac414fe37c30d/lib/Request.php#L148

But we don't actually needed to include the link and the URL encoded parameter in the mocked URL. It is what does occur on a server, but the test does still pass when removing the bit after the basepath or add a garbage string.

On that note, this is might be another case where using a `strpos !== false` instead of `strpos === 0` isn't strict enough? Impact here being that other URLs could cause this operation to get selected instead of another one (a URL that contains /shortenviayourls but not directly after the basepath *and* a link-GET-parameter) - it is the last else-if, though, so the least likely to get checked and the view operation remains the catch-all-operation.